packages/account/src/providers/utils/extract-tx-error.ts

```diff
@@ -88,8 +88,10 @@ export const assembleRevertError = (
         break;
 
       default:
-        reason = 'unknown';
-        errorMessage = `The transaction reverted with an unknown reason: ${revertReceipt.val}`;
+        throw new FuelError(
+          ErrorCode.UNKNOWN_ERROR,
+          `The transaction reverted with an unknown reason: ${revertReceipt.val}`
+        );
```

Comment:
Good point, I still believe this should be an `UNKNOWN_ERROR` but it is useful to include the metadata.

Reply:
I agree that `UNKNOWN_ERROR` can be used here as long as we use the `metadata`

Reply:
[ffa6470](https://github.com/FuelLabs/fuels-ts/pull/2562/commits/ffa64701a31bde6686bcfc5cf61894457508b4f2)

Reply:
Initially, I suggested adding this `UNKNOWN` code only for 100% untreated/unmapped errors. Those like this:

```ts
try {
  // anything that may throw
} catch (rawError) {
  throw new FuelError(
    ErrorCode.UNKNOWN,
    'Check `metadata` or `rawError` for more info.',
    { rawError },
  });
}
```

This already considered adding the `rawError` property to `FuelError`, of course.

But I'm okay with whatever the majority chooses.

Reply:
In this particular case there wouldn't be anything to include in the `rawError` property given the metadata is populated, so perhaps we should use `SCRIPT_REVERTED` or this would be an exception for an `UNKNOWN` , I lean towards the latter because even though the script is reverted, it's for an `UNKNOWN` reason.